code/swb_base/util/ScreenUtil.cs

```diff
@@ -51,7 +51,7 @@ public static void ShakeAt(Vector3 origin, float radius = 0, float delay = 0, fl
         foreach (var obj in objects)
         {
             // Player check
-            if (obj is not PlayerBase ply || !ply.IsValid())
+            if (!obj.IsValid() || obj is not ModelEntity ply)
```

Comment:
This should check for a player specifically, not ModelEntity to make it more efficient.

Reply:
`ISWBPlayer` doesn't have `PhysicsBody` as a member of it, and since there is no `IModelEntity` it becomes a slippery slope. I think my ideal here would be to refactor out the screen shake code from swb_base and move it into the example player code since screen shake effects the camera and therefore should be handled by the consumer code.

Alternatively / in addition, could we use `Entity.Position` instead of `ModelEntity.PhysicsBody.MassCenter` instead?